test/test.downscopedclient.ts

```diff
@@ -534,12 +536,100 @@ describe('DownscopedClient', () => {
     it('should throw when the source AuthClient rejects on token request', async () => {
       const expectedError = new Error('Cannot get subject token.');
       client.throwError = true;
+
       const downscopedClient = new DownscopedClient(
         client,
         testClientAccessBoundary
       );
       await assert.rejects(downscopedClient.getAccessToken(), expectedError);
     });
+
+    it('should copy source cred expiry time if STS response does not return expiry time', async () => {
+      const now = new Date().getTime();
+      const credentials = {
+        access_token: 'ACCESS_TOKEN',
+        expiry_date: now + ONE_HOUR_IN_SECS * 1000,
+      };
```

Comment:
Why do you need to set these credentials and to call `client.setCredentials(credentials);`?
I suggest removing them for this test to just focus on the expiration time population logic.

Instead you can expose a public property on `TestAuthClient` to populate the expiration time on `client.credentials.expiry_date` when `getAccessToken()` is called.

Reply:
I think there're some confusions here, getAccessToken() is contained both in downscopedClient and authClient however functions differently, in testAuthClient, it is only supposed to return the access token / subject token, the latter one is returning an object including(token, expirationTime, res). I am mocking testAuthClient class extends on authClient class but not downscopedClient.

In implementation, the actual way of getting sourceCredExpireDate() is this.authClient.credentials?.expiry_date but not calling getAccessToken(), so calling client.setCredential(credentials) is the only and built-in way of setting expiration time.